CRM/Contact/BAO/Relationship.php

```diff
@@ -63,6 +63,10 @@ public static function create(&$params) {
       throw new CRM_Core_Exception('Invalid Relationship');
     }
     $relationship = self::add($params);
+    $id = empty($params['id']) ? $relationship->id : $params['id'];
+    
+    CRM_Utils_Hook::pre('create', 'Relationship', $id, $relationship);
```

Comment:
Actually, "add" calls hooks already. So the results of this patch would be to fire the same hook twice :(